draft-ietf-acme-acme.md

```diff
@@ -1092,6 +1097,75 @@ Content-Language: en
 }
 ~~~~~
 
+### External Account Binding
+
+The server MAY require a value to be present for the "external-account-binding"
+field.  This can be used to an ACME account with an existing account in a
+non-ACME system, such as a CA customer database.
+
+To enable ACME account binding, a CA needs to provision the ACME client with a
+MAC key and a key identifier.  The key identifier MUST be an ASCII string.  The
+MAC key SHOULD be provided in base64url-encoded form, to maximize compatibility
+between provisioning systems and ACME clients.
+
+The ACME client then computes a binding JWS to indicate the external account's
+approval of the ACME account key.  The payload of this JWS is the account key
+being registered, in JWK form.  The protected header of the JWS MUST meet the
+following criteria:
+
+* The "alg" field MUST indicate a MAC-based algorithm
+* The "kid" field MUST contain the key identifier provided by the CA
+* The "nonce" field MUST NOT be present
+* The "url" field MUST be set to the same value as the outer JWS
+
+The "signature" field of the JWS will contain the MAC value computed with the
+MAC key provided by the CA.
+
+~~~~~
+POST /acme/new-reg HTTP/1.1
+Host: example.com
+Content-Type: application/jose+json
+
+{
+  "protected": base64url({
+    "alg": "ES256",
+    "jwk": /* account key */,
+    "nonce": "K60BWPrMQG9SDxBDS_xtSw",
+    "url": "https://example.com/acme/new-acct"
+  }),
+  "payload": base64url({
+    "contact": ["mailto:example@anonymous.invalid"],
+    "terms-of-service-agreed": true,
+
+    "external-account-binding": {
+      "protected": base64url({
+        "alg": "HS256",
+        "kid": /* key identifier from CA */,
+        "url": "https://example.com/acme/new-acct"
+      }),
+      "payload": base64url(/* same as in "jwk" above */),
+      "signature": /* MAC using MAC key from CA */
+    }
+  }),
+  "signature": "5TWiqIYQfIDfALQv...x9C2mg8JGPxl5bI4"
+}
+~~~~~
+
+When a CA receives a new-registration request containing an
```

Comment:
This should contain a provision allowing CAs to ignore the external-account-binding field if they do not support it. E.g.

"When a CA receives a new-registration request containing an external-account-binding field, if the CA supports that field it MUST..."